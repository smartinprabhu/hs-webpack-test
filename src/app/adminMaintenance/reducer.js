const initialState = {
  createCommentInfo: {},
  orderComments: {},
  orderParts: {},
  orderCheckLists: {},
  orderTimeSheets: {},
  priorityGroupsInfo: {},
  stateGroupsInfo: {},
  stateChangeInfo: {},
  typeGroupsInfo: {},
  workordersInfo: {},
  maintenanceInfo: {},
  // workorderCount: null,
  workorderCountErr: null,
  workorderCountLoading: false,
  maintenanceCount: null,
  maintenanceCountErr: null,
  maintenanceCountLoading: false,
  workorderDetails: {},
  maintenanceExportInfo: {},
  workorderFilters: {},
  cleaningWorkorderFilters: {},
  workorderDashboard: {},
  bookingDelete: {},
  woRows: {},
  bulkOrders: {},
  workorder: {},
  workordersCount: {},
};

// eslint-disable-next-line default-param-last
function bookingMaintenanceReducer(state = initialState, action) {
  switch (action.type) {
    case 'GET_WORK_ORDERS_INFO':
      return {
        ...state,
        workordersInfo: (state.workordersInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_INFO_SUCCESS':
      return {
        ...state,
        workordersInfo: (state.workordersInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_INFO_FAILURE':
      return {
        ...state,
        workordersInfo: (state.workordersInfo, { loading: false, err: action.error, data: null }),
      };
    // case 'GET_WORK_ORDERS_COUNT':
    //   return {
    //     ...state,
    //     workorderCountLoading: true,
    //   };
    // case 'GET_WORK_ORDERS_COUNT_SUCCESS':
    //   return {
    //     ...state,
    //     workorderCount: (state.workorderCount, action.payload.data),
    //     workorderCountLoading: false,
    //   };
    // case 'GET_WORK_ORDERS_COUNT_FAILURE':
    //   return {
    //     ...state,
    //     workorderCount: (state.workorderCount, action.error),
    //     workorderCountErr: (state.workorderCountErr, action.error),
    //     workorderCountLoading: false,
    //   };
    case 'CLEAR_MAINTENANCE_INFO':
      return {
        ...state,
        maintenanceInfo: {},
      };
    case 'GET_MAINTENANCE_INFO':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_MAINTENANCE_INFO_SUCCESS':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_MAINTENANCE_INFO_FAILURE':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_MAINTENANCE_COUNT':
      return {
        ...state,
        maintenanceCountLoading: true,
      };
    case 'GET_MAINTENANCE_COUNT_SUCCESS':
      return {
        ...state,
        maintenanceCount: (state.maintenanceCount, action.payload),
        maintenanceCountLoading: false,
      };
    case 'GET_MAINTENANCE_COUNT_FAILURE':
      return {
        ...state,
        maintenanceCount: (state.maintenanceCount, action.error),
        maintenanceCountErr: (state.maintenanceCountErr, action.error),
        maintenanceCountLoading: false,
      };
    case 'GET_WORK_ORDERS':
      return {
        ...state,
        workorder: (state.workorder, { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_SUCCESS':
      return {
        ...state,
        workorder: (state.workorder,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_FAILURE':
      return {
        ...state,
        workorder: (state.workorder, { loading: false, err: action.error, data: null }),
      };
    case 'GET_WORK_ORDERS_COUNT':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: null, data: null }),
      };
    case 'GET_WORK_ORDERS_COUNT_SUCCESS':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: null, data: action.payload }),
      };
    case 'GET_WORK_ORDERS_COUNT_FAILURE':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: action.error.data, data: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO_SUCCESS':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO_FAILURE':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: false, err: action.error, data: null }),
      };
    case 'GET_TEAMS_GROUP_INFO':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_TEAMS_GROUP_INFO_SUCCESS':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_TEAMS_GROUP_INFO_FAILURE':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATES_GROUP_INFO':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_STATES_GROUP_INFO_SUCCESS':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_STATES_GROUP_INFO_FAILURE':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO_SUCCESS':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO_FAILURE':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_ORDER_DETAILS_INFO':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: true, data: null, err: null }),
      };
    case 'GET_ORDER_DETAILS_INFO_SUCCESS':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_ORDER_DETAILS_INFO_FAILURE':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: false, err: action.error, data: null }),
      };
    case 'GET_COMMENTS_INFO':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: true, data: null, err: null }),
      };
    case 'GET_COMMENTS_INFO_SUCCESS':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_COMMENTS_INFO_FAILURE':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_COMMENTS_INFO':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, err: null, data: null }),
      };
    case 'CREATE_COMMENT_INFO':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: true, data: null, err: null }),
      };
    case 'CREATE_COMMENT_INFO_SUCCESS':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'CREATE_COMMENT_INFO_FAILURE':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATE_CHANGE_INFO':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_STATE_CHANGE_INFO_SUCCESS':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_STATE_CHANGE_INFO_FAILURE':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATE_RESET_INFO':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, data: null, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: true, data: null, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO_SUCCESS':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO_FAILURE':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: false, err: action.error, data: null }),
      };
    case 'CLEAR_CLEANING_WORKORDER_FILTERS':
      return {
        ...state,
        cleaningWorkorderFilters: {},
      };
    case 'GET_PARTS_INFO':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: true, data: null, err: null }),
      };
    case 'GET_PARTS_INFO_SUCCESS':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_PARTS_INFO_FAILURE':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: false, err: action.error, data: null }),
      };
    case 'GET_CHECK_LISTS_INFO':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: true, data: null, err: null }),
      };
    case 'GET_CHECK_LISTS_INFO_SUCCESS':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_CHECK_LISTS_INFO_FAILURE':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: false, err: action.error, data: null }),
      };
    case 'GET_TIMESHEETS_INFO':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: true, data: null, err: null }),
      };
    case 'GET_TIMESHEETS_INFO_SUCCESS':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_TIMESHEETS_INFO_FAILURE':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: false, err: action.error, data: null }),
      };
    case 'CLEAR_BOOKINGS_FILTER_WORKORDER':
      return {
        ...state,
        workorderFilters: {},
      };
    case 'CLEAR_BOOKINGS_WORKORDER_COUNT':
      return {
        ...state,
        maintenanceCount: {},
      };
    case 'GET_FILTER_WORKORDER':
      return {
        ...state,
        workorderFilters: (state.workorderFilters, action.payload),
      };
    case 'GET_FILTER_CLEANING_WORKORDER':
      return {
        ...state,
        cleaningWorkorderFilters: (state.cleaningWorkorderFilters, action.payload),
      };
    case 'GET_ROWS_WO':
      return {
        ...state,
        woRows: (state.woRows, action.payload),
      };
    case 'DELETE_BOOKING':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: true, data: null, err: null }),
      };
    case 'DELETE_BOOKING_SUCCESS':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, data: action.payload.data, err: null }),
      };
    case 'DELETE_BOOKING_FAILURE':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, err: action.error.data, data: null }),
      };
    case 'SAVE_BULK_ORDERS':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: true, data: null, err: null }),
      };
    case 'SAVE_BULK_ORDERS_SUCCESS':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, data: action.payload.data, err: null }),
      };
    case 'SAVE_BULK_ORDERS_FAILURE':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, err: action.error.data, data: null }),
      };
    case 'CLEAR_CLEANING':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, data: null, err: null }),
      };
    case 'CLEAR_BOOKING':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, data: null, err: null }),
      };
    default:
      return state;
  }
}

export default bookingMaintenanceReducer;
