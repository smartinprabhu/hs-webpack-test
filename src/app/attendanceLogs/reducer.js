/* eslint-disable linebreak-style */
const initialState = {
  attendanceLogs: {},
  attendanceCount: {},
  attendanceExportInfo: {},
  attendanceFiltersInfo: {},
  attendanceReportFiltersInfo: {},
  attendanceDetailsInfo: {},
  createExportInfo: {},
  exportLinkInfo: {},
  attendanceCountLoading: false,
  attendanceCountErr: null,
};
function reducer(state, action) {
  if (typeof state === 'undefined') {
    return initialState;
  }
  switch (action.type) {
    case 'GET_ATTENDANCE_DATA':
      return {
        ...state,
        attendanceLogs: (state.attendanceLogs, { loading: true, data: null, err: null }),
      };
    case 'GET_ATTENDANCE_DATA_SUCCESS':
      return {
        ...state,
        attendanceLogs: (state.attendanceLogs, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_ATTENDANCE_DATA_FAILURE':
      return {
        ...state,
        attendanceLogs: (state.attendanceLogs, { loading: false, err: action.error, data: null }),
      };
    case 'GET_ATTENDANCE_COUNT':
      return {
        ...state,
        attendanceCountLoading: true,
      };
    case 'GET_ATTENDANCE_COUNT_SUCCESS':
      return {
        ...state,
        attendanceCount: (state.attendanceCount, action.payload),
        attendanceCountLoading: false,
      };
    case 'GET_ATTENDANCE_COUNT_FAILURE':
      return {
        ...state,
        attendanceCountErr: (state.attendanceCountErr, action.error),
        attendanceCountLoading: false,
      };
    case 'GET_ATTEDANCE_EXPORT':
      return {
        ...state,
        attendanceExportInfo: (state.attendanceExportInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_ATTEDANCE_EXPORT_SUCCESS':
      return {
        ...state,
        attendanceExportInfo: (state.attendanceLogs, { loading: false, data: action.payload.data, err: null }),

      };
    case 'GET_ATTEDANCE_EXPORT_FAILURE':
      return {
        ...state,
        attendanceExportInfo: (state.attendanceExportInfo, { loading: false, err: action.error, data: null }),
      };
    case 'ATTENDANCE_FILTERS':
      return {
        ...state,
        attendanceFiltersInfo: (state.attendanceFiltersInfo, action.payload),
      };
    case 'GET_ATTEDANCE_DETAILS':
      return {
        ...state,
        attendanceDetailsInfo: (state.attendanceDetailsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_ATTEDANCE_DETAILS_SUCCESS':
      return {
        ...state,
        attendanceDetailsInfo: (state.attendanceDetailsInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_ATTEDANCE_DETAILS_FAILURE':
      return {
        ...state,
        attendanceDetailsInfo: (state.attendanceDetailsInfo, { loading: false, data: null, err: action.error }),
      };
    case 'CREATE_ATT_EXPORT_INFO':
      return {
        ...state,
        createExportInfo: (state.createExportInfo, { loading: true, data: null, err: null }),
      };
    case 'CREATE_ATT_EXPORT_INFO_SUCCESS':
      return {
        ...state,
        createExportInfo: (state.createExportInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'CREATE_ATT_EXPORT_INFO_FAILURE':
      return {
        ...state,
        createExportInfo: (state.createExportInfo, { loading: false, data: null, err: action.error }),
      };
    case 'RESET_CREATE_ATT_EXPORT_INFO':
      return {
        ...state,
        createExportInfo: (state.createExportInfo, { loading: false, data: null, err: null }),
      };
    case 'GET_EXPORT_LINK_INFO':
      return {
        ...state,
        exportLinkInfo: (state.exportLinkInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_EXPORT_LINK_INFO_SUCCESS':
      return {
        ...state,
        exportLinkInfo: (state.exportLinkInfo, { loading: false, data: action.payload, err: null }),
      };
    case 'GET_EXPORT_LINK_INFO_FAILURE':
      return {
        ...state,
        exportLinkInfo: (state.exportLinkInfo, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_EXPORT_LINK_INFO':
      return {
        ...state,
        exportLinkInfo: (state.exportLinkInfo, { loading: false, err: null, data: null }),
      };
    case 'ATTENDANCE_REPORT':
      return {
        ...state,
        attendanceReportFiltersInfo: action.payload,
      };
    default:
      return state;
  }
}

export default reducer;
