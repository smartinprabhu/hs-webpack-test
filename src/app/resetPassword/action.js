import { CALL_API } from '../../middleware/api';

export const RESET_PASSWORD_SUCCESS = 'RESET_PASSWORD_SUCCESS';
export const RESET_PASSWORD_FAILURE = 'RESET_PASSWORD_FAILURE';
export const RESET_PASSWORD = 'RESET_PASSWORD';

export const GET_PASSWORD_POLICY_SUCCESS = 'GET_PASSWORD_POLICY_SUCCESS';
export const GET_PASSWORD_POLICY_FAILURE = 'GET_PASSWORD_POLICY_FAILURE';
export const GET_PASSWORD_POLICY = 'GET_PASSWORD_POLICY';

export const RESET_PASSWORD_LINK_SUCCESS = 'RESET_PASSWORD_LINK_SUCCESS';
export const RESET_PASSWORD_LINK_FAILURE = 'RESET_PASSWORD_LINK_FAILURE';
export const RESET_PASSWORD_LINK = 'RESET_PASSWORD_LINK';

export const CHECK_RESET_PASSWORD_LINK = 'CHECK_RESET_PASSWORD_LINK';
export const CHECK_RESET_PASSWORD_LINK_SUCCESS = 'CHECK_RESET_PASSWORD_LINK_SUCCESS';
export const CHECK_RESET_PASSWORD_LINK_FAILURE = 'CHECK_RESET_PASSWORD_LINK_FAILURE';

export const SAVE_SESSIONUUID = 'SAVE_SESSIONUUID';
export const CLEAR_RESET_PASSWORD_DATA = 'CLEAR_RESET_PASSWORD_DATA';
export const SHOW_RESET_PASSWORD = 'SHOW_RESET_PASSWORD';

// to sent reset password link

export function getResetPasswordLink(data) {
  return {
    [CALL_API]: {
      endpoint: 'user/forgot_password',
      types: [
        RESET_PASSWORD_LINK, RESET_PASSWORD_LINK_SUCCESS, RESET_PASSWORD_LINK_FAILURE,
      ],
      method: 'POST',
      payload: data,
    },
  };
}

// to get reset password policy

export function getPasswordPolicy() {
  return {
    [CALL_API]: {
      endpoint: 'user/password_policy?token=45g1FGWa8ILBJQl9sNEaevmYo2TgYP',
      types: [
        GET_PASSWORD_POLICY,
        GET_PASSWORD_POLICY_SUCCESS,
        GET_PASSWORD_POLICY_FAILURE,
      ],
      method: 'GET',
    },
  };
}

// to check check_reset_password_link

export function checkResetPasswordLink(values) {
  return {
    [CALL_API]: {
      endpoint: 'user/check_reset_password_link',
      types: [
        CHECK_RESET_PASSWORD_LINK,
        CHECK_RESET_PASSWORD_LINK_SUCCESS,
        CHECK_RESET_PASSWORD_LINK_FAILURE,
      ],
      method: 'POST',
      payload: values,
    },
  };
}

// to reset password

export function resetPassword(data) {
  return {
    [CALL_API]: {
      endpoint: 'user/reset_password',
      types: [
        RESET_PASSWORD,
        RESET_PASSWORD_SUCCESS,
        RESET_PASSWORD_FAILURE,
      ],
      method: 'POST',
      payload: data,
    },
  };
}

// to save session_uuid

export function saveSessionId(value) {
  return {
    type: SAVE_SESSIONUUID,
    payload: value,
  };
}

export function clearResetPassowrdRes() {
  return {
    type: CLEAR_RESET_PASSWORD_DATA,
  };
}

export function showForgotMailFormData() {
  return {
    type: CLEAR_RESET_PASSWORD_DATA,
  };
}

export function showResetPasswordFormData() {
  return {
    type: SHOW_RESET_PASSWORD,
  };
}
