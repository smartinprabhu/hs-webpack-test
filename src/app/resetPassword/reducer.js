import {
  CLEAR_RESET_PASSWORD_DATA, SHOW_RESET_PASSWORD,
} from './action';

const initialState = {
  checkResetPasswordLinkInfo: {},
  passwordPrivacyPolicyInfo: {},
  resetPasswordInfo: {},
  resetPasswordLinkInfo: {},
  session: null,
};
function reducer(state, action) {
  if (typeof state === 'undefined') {
    return initialState
  }
  switch (action.type) {
    case 'RESET_PASSWORD':
      return {
        ...state,
        resetPasswordInfo: (state.resetPasswordInfo, { loading: true })
      };
    case 'RESET_PASSWORD_SUCCESS':
      return {
        ...state,
        resetPasswordInfo: (state.resetPasswordInfo, { loading: false, data: action.payload.data })
      };
    case 'RESET_PASSWORD_FAILURE':
      return {
        ...state,
        resetPasswordInfo: (state.resetPasswordInfo, { loading: false, err: action.error.data })
      };
    case 'GET_PASSWORD_POLICY':
      return {
        ...state,
        passwordPrivacyPolicyInfo: (state.passwordPrivacyPolicyInfo, { loading: true })
      };
    case 'GET_PASSWORD_POLICY_SUCCESS':
      return {
        ...state,
        passwordPrivacyPolicyInfo: (state.passwordPrivacyPolicyInfo, { loading: false, data: action.payload })
      };
    case 'GET_PASSWORD_POLICY_FAILURE':
      return {
        ...state,
        passwordPrivacyPolicyInfo: (state.passwordPrivacyPolicyInfo, { loading: false, err: action.error.data })
      };
    case 'RESET_PASSWORD_LINK':
      return {
        ...state,
        resetPasswordLinkInfo: (state.resetPasswordLinkInfo, { loading: true })
      };
    case 'RESET_PASSWORD_LINK_SUCCESS':
      return {
        ...state,
        resetPasswordLinkInfo:
          (state.resetPasswordLinkInfo, { loading: false, data: action.payload.data })
      };
    case 'RESET_PASSWORD_LINK_FAILURE':
      return {
        ...state,
        resetPasswordLinkInfo: (state.resetPasswordLinkInfo, { loading: false, err: action.error.data })
      };
    case 'CHECK_RESET_PASSWORD_LINK':
      return {
        ...state,
        checkResetPasswordLinkInfo: (state.checkResetPasswordLinkInfo, { loading: true })
      };
    case 'CHECK_RESET_PASSWORD_LINK_SUCCESS':
      return {
        ...state,
        checkResetPasswordLinkInfo: (state.checkResetPasswordLinkInfo, { loading: false, data: action.status })
      };
    case 'CHECK_RESET_PASSWORD_LINK_FAILURE':
      return {
        ...state,
        checkResetPasswordLinkInfo: (state.checkResetPasswordLinkInfo, { loading: false, err: action.error.data })
      };
    case 'SAVE_SESSIONUUID':
      return {
        ...state,
        session: action.payload
      };

    case CLEAR_RESET_PASSWORD_DATA:
      return {
        ...state,
        resetPasswordLinkInfo: (state.resetPasswordLinkInfo, { loading: false, data: null, err: null })
      };

    case SHOW_RESET_PASSWORD:
      return {
        ...state,
        resetPasswordInfo: (state.resetPasswordInfo, {
          loading: false, data: null, err: null, passwordPrivacyPolicyInfo: null
        })
      };

    default: return state;
  }
}
export default reducer;
