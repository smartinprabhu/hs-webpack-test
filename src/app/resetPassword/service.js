import {
  resetPassword, checkResetPasswordLink, getPasswordPolicy, getResetPasswordLink, clearResetPassowrdRes, showForgotMailFormData, showResetPasswordFormData,
} from './action';

export function createNewPassword(data) {
  return (dispatch) => {
    dispatch(resetPassword(data));
  };
}
export function testResetPasswordLink(values) {
  return (dispatch) => {
    dispatch(checkResetPasswordLink(values));
  };
}
export function getPasswordPolicyInfo() {
  return (dispatch) => {
    dispatch(getPasswordPolicy());
  };
}
export function sendResetPasswordLink(data) {
  return (dispatch) => {
    dispatch(getResetPasswordLink(data));
  };
}

export function clearResetPasswordData() {
  return (dispatch) => {
    dispatch(clearResetPassowrdRes());
  };
}

export function showForgotMailForm() {
  return (dispatch) => {
    dispatch(showForgotMailFormData());
  };
}

export function showResetPasswordForm() {
  return (dispatch) => {
    dispatch(showResetPasswordFormData());
  };
}
