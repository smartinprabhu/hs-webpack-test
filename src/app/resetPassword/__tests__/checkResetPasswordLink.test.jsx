import React from 'react';
import * as redux from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import { cleanup, render, fireEvent } from '@testing-library/react';

import api from '../../../middleware/api';
import * as actions from '../action';
import * as services from '../service';
import reducer from '../reducer';
import CheckResetPasswordLink from '../checkResetPasswordLink';

const mockHistoryPush = jest.fn();

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

jest.mock('react-redux', () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => jest.fn(),
}));

afterEach(cleanup);

describe('checkResetPasswordLink testing', () => {
  it('snapshot testing', () => {
    redux.useSelector.mockImplementation((fn) => fn({
      resetPassword: {
        checkResetPasswordLinkInfo: {
          loading: true,
        },
      },
    }));
    const { asFragment } = render(
      <CheckResetPasswordLink location={{ session_uuid: '?uuid=594ebbd6aef01565b1c43c66e04c1bc0c2126c9a4771cdcd46d77b7a5ab4a931:63dc7ed1010d3c3b8269faf0ba7491d4' }} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('it should mock checkResetPasswordLink function', () => {
    const checkResetPasswordLink = (values) => actions.checkResetPasswordLink(values);
    checkResetPasswordLink();
    expect(actions.checkResetPasswordLink).toBeDefined();
  });

  it('it should create an action to save session_uuid', () => {
    const sessionUuid = '1234:4567';
    const expectedAction = {
      type: actions.SAVE_SESSIONUUID,
      payload: sessionUuid,
    };
    expect(actions.saveSessionId(sessionUuid)).toEqual(expectedAction);
  });

  it('it should mock the services', () => {
    const testResetPasswordLink = jest.spyOn(services, 'testResetPasswordLink');
    services.testResetPasswordLink();
    expect(testResetPasswordLink).toBeCalled();
  });

  const initialState = {
    checkResetPasswordLinkInfo: {},
    passwordPrivacyPolicyInfo: {},
    resetPasswordInfo: {},
    resetPasswordLinkInfo: {},
    session: null,
  };

  it('it should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState);
  });

  it('it should return the CHECK_RESET_PASSWORD_LINK', () => {
    expect(reducer(initialState, {
      type: actions.CHECK_RESET_PASSWORD_LINK,
    })).toEqual({
      checkResetPasswordLinkInfo: { loading: true },
      passwordPrivacyPolicyInfo: {},
      resetPasswordInfo: {},
      resetPasswordLinkInfo: {},
      session: null,
    });
  });

  it('it should return the CHECK_RESET_PASSWORD_LINK_SUCCESS', () => {
    expect(reducer(initialState, {
      type: actions.CHECK_RESET_PASSWORD_LINK_SUCCESS,
      status: true,
    })).toEqual({
      checkResetPasswordLinkInfo: { loading: false, data: true },
      passwordPrivacyPolicyInfo: {},
      resetPasswordInfo: {},
      resetPasswordLinkInfo: {},
      session: null,
    });
  });

  it('it should return the CHECK_RESET_PASSWORD_LINK_FAILURE', () => {
    expect(reducer(initialState, {
      type: actions.CHECK_RESET_PASSWORD_LINK_FAILURE,
      error: {
        data: {
          message: 'Session expired or invalid.',
        },
      },
    })).toEqual({
      checkResetPasswordLinkInfo: {
        loading: false,
        err: {
          message: 'Session expired or invalid.',
        },
      },
      passwordPrivacyPolicyInfo: {},
      resetPasswordInfo: {},
      resetPasswordLinkInfo: {},
      session: null,
    });
  });

  it('it should return the SAVE_SESSIONUUID', () => {
    expect(reducer(initialState, {
      type: actions.SAVE_SESSIONUUID,
      payload: {},
    })).toEqual({
      checkResetPasswordLinkInfo: {},
      passwordPrivacyPolicyInfo: {},
      resetPasswordInfo: {},
      resetPasswordLinkInfo: {},
      session: {},
    });
  });

  const create = () => {
    const store = jest.fn();
    const next = jest.fn();
    const invoke = (action) => api(store)(next)(action);
    return { next, invoke };
  };

  it('passes through CHECK_RESET_PASSWORD_LINK action', () => {
    const { next, invoke } = create();
    const action = { type: actions.CHECK_RESET_PASSWORD_LINK };
    invoke(action);
    expect(next).toHaveBeenCalledWith(action);
  });

  it('dom testing', () => {
    redux.useSelector.mockImplementation((fn) => fn({
      resetPassword: {
        checkResetPasswordLinkInfo: {
          err: {
            error: {
              message: 'Session expired or invalid.',
            },
          },
        },
      },
    }));

    const { container, getByText } = render(<CheckResetPasswordLink
      location={{ session_uuid: '?uuid=594ebbd6aef01565b1c43c66e04c1bc0c2126c9a4771cdcd46d77b7a5ab4a931:63dc7ed1010d3c3b8269faf0ba7491d4' }}
    />);
    expect(getByText('Session expired or invalid.')).toBeInTheDocument();
    expect(getByText('Forgot Password')).toBeInTheDocument();
    expect(container).toBeInTheDocument();
  });
  it('dom testing', () => {
    redux.useSelector.mockImplementation((fn) => fn({
      resetPassword: {
        checkResetPasswordLinkInfo: {
          data: true,
          loading: true,
        },
      },
    }));
    const { getByText, getByRole } = render(<CheckResetPasswordLink
      location={{ session_uuid: '?uuid=594ebbd6aef01565b1c43c66e04c1bc0c2126c9a4771cdcd46d77b7a5ab4a931:63dc7ed1010d3c3b8269faf0ba7491d4' }}
    />);
    expect(getByRole('button')).toBeInTheDocument();
    fireEvent.click(getByText('Reset Password'));
    expect(mockHistoryPush).toBeCalledWith({ pathname: '/create-new-password' });
  });
});
