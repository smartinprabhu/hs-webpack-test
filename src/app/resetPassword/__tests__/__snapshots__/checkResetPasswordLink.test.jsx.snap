// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkResetPasswordLink testing snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="login"
  >
    <div
      class="m-0 row"
    >
      <div
        class="bg-white p-4 mt-7 opacity-9 col-sm-4 offset-sm-7"
      >
        <div
          class="m-0 row"
        >
          <div
            class="pl-0 col-sm-9"
          >
            <h3>
              Forgot Password
            </h3>
          </div>
          <div
            class="col-sm-3"
          >
            <img
              alt="Helixsense Portal"
              class="float-right"
              src="test-file-stub"
              width="120"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
